openmls/src/treesync/node/leaf_node.rs

```diff
@@ -382,6 +384,26 @@ impl TlsDeserializeTrait for LeafNodeTbs {
     }
 }
 
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsDeserialize)]
+pub struct StateUnvalidated;
+
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsSerialize)]
+pub struct StateKeyPackage;
+
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsSerialize)]
+pub struct StateAddUpdateCommit;
+
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsSerialize)]
+pub struct StateRatchetTree;
+
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsSerialize)]
+pub struct StateValidated;
```

Comment:
This state can be created from any of the three other "validated" states. The reason we have this is because we use the leaf node "generically" as soon as we have it in the ratchet tree. I'm not sure if this is tight enough or if we loose guarantees due to this. In any case, I don't like this and have to think again.